### PR TITLE
{Network} Add missing commands for `client-cert` and `ssl-profile`

### PR DESCRIPTION
**Pull request by 00Kai0**
closed https://github.com/Azure/azure-cli/issues/19036
**Description**

**Testing Guide**

**History Notes**

[Network] `az network application-gateway client-cert`: add commands `update` and `show`
[Network] `az network application-gateway ssl-profile`: add commands `update` and `show`
[Network] `az network application-gateway http-listener create`: add parameter `--ssl-profile`
[Network] `az network application-gateway http-listener update`: add parameter `--ssl-profile`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 1]
```diff
@@ -1542,6 +1558,14 @@
     text: az network application-gateway ssl-profile add --gateway-name MyAppGateway -g MyResourceGroup --name MySslProfile
 """
 
+helps['network application-gateway ssl-profile update'] = """
+type: command
+short-summary: Update ssl profiles of the application gateway.
+examples:
+  - name: Add ssl profile for an existing application gateway.
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:1565]:
```suggestion
  - name: Update ssl profile for an existing application gateway.
```

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -1550,6 +1574,14 @@
     text: az network application-gateway ssl-profile remove --gateway-name MyAppGateway -g MyResourceGroup --name MySslProfile
 """
 
+helps['network application-gateway ssl-profile show'] = """
+type: command
+short-summary: Show an existing ssl profiles of the application gateway.
+examples:
+  - name: Remove ssl profile for an existing application gateway.
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:1581]:
```suggestion
  - name: Show ssl profile for an existing application gateway.
```

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 3]
```diff
@@ -1521,6 +1529,14 @@
     text: az network application-gateway client-cert remove --gateway-name MyAppGateway -g MyResourceGroup --name MyCert
 """
 
+helps['network application-gateway client-cert show'] = """
+type: command
+short-summary: Show an existing trusted client certificate of the application gateway.
+examples:
+  - name: Remove a trusted client certificate for an existing application gateway.
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:1536]:
```suggestion
  - name: Show a trusted client certificate for an existing application gateway.
```

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 4]
```diff
@@ -1513,6 +1513,14 @@
     text: az network application-gateway client-cert add --gateway-name MyAppGateway -g MyResourceGroup --name MyCert --data Cert.cer
 """
 
+helps['network application-gateway client-cert update'] = """
+type: command
+short-summary: Update trusted client certificate of the application gateway.
+examples:
+  - name: Add trusted client certificate for an existing application gateway.
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:1520]:
```suggestion
  - name: Update trusted client certificate for an existing application gateway.
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 5]
```diff
@@ -384,7 +384,7 @@ def update_ag_frontend_port(instance, parent, item_name, port=None):
 
 def create_ag_http_listener(cmd, resource_group_name, application_gateway_name, item_name,
                             frontend_port, frontend_ip=None, host_name=None, ssl_cert=None,
-                            firewall_policy=None, no_wait=False, host_names=None):
+                            ssl_profile=None, firewall_policy=None, no_wait=False, host_names=None):
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:387]: Do you have any test for this property?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**